greymatter-io/operator
# add in logic for the operator to not copy docker secret accross names…

…paces


This is a companion pr to https://github.com/greymatter-io/gitops-core/pull/53 which adds a config to gitops core.

`auto_copy_image_pull_secret` settings:

`true` will have the standard feature which will copy the image pull secret from the gm-operator namespace into the install namespace as well as any watch namespaces
`false` will not copy the image pull secret. The operator does log if it can not find the secret.

The operator will still create all resources required for greymatter core however these will fail to start until an image pull of name `gm-docker-secret` is added to the namespace.

## Prerequisite

- Build and push your own image `scripts/build container && docker tag docker.greymatter.io/internal/gm-operator:latest quay.io/greymatterio/operator:mytest && docker push quay.io/greymatterio/operator:mytest`
- update your operator statefulset manifest to use the image `quay.io/greymatterio/operator:mytest`
- create a branch of gitops core based on this [branch](https://github.com/greymatter-io/gitops-core/tree/no-auto-apply-image-pull-secret)
- update your operator statefulset manifest to use your branch

## Test Normal

- Setup a normal mesh following the playbook.
- Success: Everything should operate like normal.
- You should see the `gm-docker-secret` in the `greymatter` namespace and all watched namespaces.

Cleanup: by deleting the operator and letting all resources get destroyed.  Delete the `greymatter` namespace and all watched namespaces and wait for it to fully terminate.

## Test new flag

- In your gitops-core `inputs.cue` file set `auto_copy_image_pull_secret settings` to `false`
- launch the operator
- The operator should be installed and start creating resources
- You will see pods being created in the `greymatter` namespace however they will not start since there is no docker pull secret
- Apply your docker image pull secret in the `greymatter` name space.
- The kubernetes pod reconciliation loop will start them up in the next couple min.
- check the secrets in all watched namespaces.  You should not see the secret copied into other namespaces.

Cleanup: same as before
